### PR TITLE
Adds "Save Song" button to save music to localstorage.Still in need…

### PR DESCRIPTION
**Pull request by scoyne122**
Added a "save song" button that saves a song to local storage. Still needs a "my music" section that can display the users saved songs.

### CODE REVIEW

File: js/everything.js [hunk 1]
```diff
@@ -261,8 +262,21 @@ var ZenPlayer = {
...
+    		var mySongs = JSON.parse(localStorage.getItem("mySongs"));
+    		if (mySongs === null) {
+    			localStorage.setItem("mySongs", JSON.stringify([getCurrentVideoID()]));
+    		}
+    		else {
+    			mySongs.push(getCurrentVideoID());
+    			localStorage.setItem("mySongs", JSON.stringify(mySongs));
+    		}
+    	});
+    }
```
Comment (eirism) [inline js/everything.js]: Insert a comma here to see if it builds

Comment (eirism) [inline js/everything.js]: Insert a comma here and see if it builds

File: index.html [hunk 2]
```diff
@@ -82,7 +82,9 @@ <h2 class="color-grey">Listen to YouTube videos, without the distracting visuals
             </div>
             <br>
             <div id="zen-video-description"></div>
+	    <div id="save" class="btn">Save Song</div>
             <br>
+	    <br>
```
Comment (shakeelmohamed) [inline index.html]: why this added break?

File: js/everything.js [hunk 3]
```diff
@@ -169,6 +169,7 @@ var ZenPlayer = {
                 that.setupTitle();
                 that.setupVideoDescription();
                 that.setupPlyrToggle();
+		that.setupSongSave();
```
Comment (shakeelmohamed) [inline js/everything.js]: tab indents

File: js/everything.js [hunk 4]
```diff
@@ -261,8 +262,21 @@ var ZenPlayer = {
         // Show player button click event
         $("#togglePlayer").click(function(event) {
             toggleElement(event, ".plyr__video-wrapper", "Player");
+            	
         });
     },
+    setupSongSave: function() {
+    	$("#save").click(function() {
+    		var mySongs = JSON.parse(localStorage.getItem("mySongs"));
```
Comment (shakeelmohamed) [inline js/everything.js]: tab

File: js/everything.js [hunk 5]
```diff
@@ -261,8 +262,21 @@ var ZenPlayer = {
         // Show player button click event
         $("#togglePlayer").click(function(event) {
             toggleElement(event, ".plyr__video-wrapper", "Player");
+
```
Comment (shakeelmohamed) [inline js/everything.js]: revert this change

File: js/everything.js [hunk 6]
```diff
@@ -261,8 +262,21 @@ var ZenPlayer = {
         // Show player button click event
         $("#togglePlayer").click(function(event) {
             toggleElement(event, ".plyr__video-wrapper", "Player");
+            	
         });
     },
+    setupSongSave: function() {
+    	$("#save").click(function() {
+    		var mySongs = JSON.parse(localStorage.getItem("mySongs"));
+    		if (mySongs === null) {
```
Comment (shakeelmohamed) [inline js/everything.js]: lots of tabs here 😄

**Comment by shakeelmohamed** [comment]
@scoyne122 thanks for your efforts here. I think this raises an interesting idea, I've been thinking about doing something similar. The "right" way to implement this would be using the YouTube API (otherwise you'll lose your list when changing browsers or devices) to save videos to a playlist, and require oauth (see what I did for [YouTube watch later](https://github.com/shakeelmohamed/youtube-watch-later)). This also ties into #1 - adding support for YouTube playlists.

However, it's not quite that simple. When #186 goes in, we'll have Soundcloud support, meaning we'd need to support and cache 2 auth tokens. That's not necessarily a bad thing, but something we should collectively think about. Then we should consider how we want to handle playlists: a) YouTube & Soundcloud playlists are separated, or b) we add our own playlists feature that can contain music from either source. I think option b is more use friendly, but will be trickier to implement.

Let me know what you guys think! cc: @zen-audio-player/contributors

**Comment by scoyne122** [comment]
@shakeelmohamed I agree with moving to a cloud based approach to ensure the user's music transfers over devices or browsers. However, assuming Sound Cloud is integrated, it may be easier to implement our own user login system rather than the youtube API with oauth. Perhaps we could create a PHP or node.js backend that connects to a mySQL?

Let me know your thoughts.

**Comment by shakeelmohamed** [comment]
> Perhaps we could create a PHP or node.js backend that connects to a mySQL?

I've been against this so far in order to keep the project both free and totally open. What you see on the `master` branch of this repo is exactly what's served on https://ZenPlayer.Audio since we're using GitHub pages. Even if we implemented our own accounts, they would be based on oauth from another platform.  I'm not interested in adding a database for this project because of the added cost and complexity. If anyone needs to touch the backend or database there's additional security issues that will arise.

**Comment by scoyne122** [comment]
Fair enough.
If we do decide on adding our own playlist feature (which I'm still a fan of), what would be your preferred method of doing so?

**Comment by shakeelmohamed** [comment]
We could do something like having a reserved playlist name ("Zen Audio Player") on both YouTube and Soundcloud, then aggregate them on the website depending on which service(s) the user is authenticated.

**Comment by YuriBrunetto** [comment]
Hey guys! 
@scoyne122 great work 👍 

But, as @shakeelmohamed said: I also think that it would be better to create a playlist on YouTube (maybe something with the name like `Yuri's ZenAudio Playlist`).

**Comment by shakeelmohamed** [comment]
@scoyne122 shall we close this PR?

**Comment by shakeelmohamed** [comment]
Closing since we've decided this won't be merged. Let's continue this conversation in #229
